src/unify/mod.rs

```diff
@@ -232,17 +243,39 @@ impl<K: UnifyKey> VarValue<K> {
         }
     }
 }
+impl<K, V, L> UnificationTable<InPlace<K, V, L>>
+where
+    K: UnifyKey,
+    V: sv::VecLike<Delegate<K>>,
+{
+    pub fn with_log<'a, L2>(
```

Comment:
Can you document how this method is meant to be used and what it does?

It seems a bit surprising to me since ordinarily I think of there being "just one" undo-log for a given table. 

I guess the idea here is that, since snapshots are "stacked", you can "push a new log" on and you'll fully work through the snapshots that are in that new log before you come back to whatever log was there before?

Reply:
Changed this to only be implemented on the `Storage` types which has `()` in place of the undo log. Since `()` does not implement `UndoLogs` mutating methods can't be called. `with_log` creates an actual `UnificationTable` which can then be mutated.